### ISSUE TITLE
Tracks not scrobbled after seeking

### ISSUE THREAD

**Issue by sjuvonen** [clementine-player/Clementine#4836]
Last.fm scrobbling fails almost every single time if the track is being seeked into any direction. This isn't a simple user error regarding playing the track long enough, even backwards seeking breaks submissions. From console output it's pretty clear Clementine does not try to scrobble these tracks since there is no reply printed out either.

**Comment by vrga** [comment]
Isnt this how the scrobbler is supposed to work, to avoid gaming the system? Any scroll in song === no scrobble.

**Comment by sjuvonen** [comment]
How would seeking "game the system"? To my mind, the scrobble counter should run independently of track progress; just count how many seconds the track has been played for and once it hits the limit -> scrobble. It shouldn't matter which segment of the track is actually played. You still have to use the same amount of time to score a scrobble.

**Comment by vrga** [comment]
http://www.last.fm/api/scrobbling#when-is-a-scrobble-a-scrobble

i think it must still be uninterrupted 4 minutes minimum...

**Comment by sjuvonen** [comment]
Yes, I've read that a few times since I work with Last.fm scrobbling too. There are no special requirements for "has been played for at least half of its duration", so to me it is sufficient that the track has been playing for a certain amount of time in one way or another.

**Comment by vrga** [comment]
https://github.com/clementine-player/Clementine/blob/master/src/playlist/playlist.cpp#L1708 seems to be the function that calculates the scrobble status.

**Comment by vrga** [comment]
I really dont want to try tracing this further as i dont have the knowledge to actually do stuff with qt.

this was just figured out through a bit of grep, luck and general tracing knowhow...

I could also be completely off the mark.

**Comment by TheUbuntuGuy** [comment]
I've seen the code in the engine that does this and thought about fixing this problem about a year ago. It's a bit messy though; everything is zero-based, so adding support for seeking could cause a lot of regressions. I agree that it should still scrobble if you listen to 50% of the track in length or 4 minutes, but it shouldn't have to be from the beginning. I'll think about trying a patch soon.
